apis/python/src/tiledbsoma/dataframe.py

```diff
@@ -41,6 +41,7 @@ def create(
         *,
         schema: pa.Schema,
         index_column_names: Sequence[str] = (SOMA_JOINID,),
+        shape: Optional[int] = None,
```

Comment:
This signature (and the docstring below) do not handle cases where the indexed columns (aka dimensions) are non-integer.  How are (for example) date/time and float handled?

Reply:
This also pointed out a major gap in the DataFrame unit test -- there is no test of indexing on column types other than int and string.

As I understand the spec, any user-defined dataframe column can be indexed, which means we need tests for float and other (non-string, non-int) types.

Reply:
Additional issue - there can be more than one index column on a DataFrame.  So shouldn't the capacity be a sequence?

Reply:
Added #956 956 on the issue of poor unit test coverage, which IMHO would help shake out this PR.

Reply:
As of the most recent commit, this PR retracts any attempts at explicit shaping of `DataFrame`. Those questions will now be addressed on a separate PR.